### PR TITLE
Day and night cycle!

### PR DESCRIPTION
**Pull request by Gandalf2k15**
## About The Pull Request

Are you bored of the same old constant nighttime? Worry not!

The day/night system adds a way for people to tell what the fucking time is while making sense. It also adds a cool system for controlling the brightness of the outside on say icebox so it actually feels like a planet with a sun etc.

Currently, every minute that passes is equal to 10 minutes for the station time.

(video is sped up lol)
https://user-images.githubusercontent.com/9026500/177439068-46324f0f-0d38-4fb4-b83c-b21a36ba5951.mp4

This system uses presets and timezones to render a nice setting and give people an idea of how much time passes.

TO-DO:
- [ ] Add the super optimised mutable appearance over areas instead of the set_light system.
- [ ] Add a watch with the ability to set alarms and display the current time when worn as maptext.
- [ ] Add solar power generation differentials.
- [ ] Add parallax shifting for space.
- [ ] Add day/night lighting integration.

This also adds two new quirks, daytime lover and nighttime lover.
These basically give slight moodlets to being outside during a specific time, and negative moodlets for being out at a specific time.

## Changelog

:cl:
add: A new day/night cycle to the game!
add: A new daytime lover quirk!
add: A new nighttime lover quirk!
/:cl:

### CODE REVIEW

**Comment by azrael-x86** [comment]
why not upstream

File: code/controllers/subsystem/statpanel.dm [hunk 1]
```diff
@@ -46,7 +46,7 @@ SUBSYSTEM_DEF(statpanels)
 			"OOC: [GLOB.ooc_allowed ? "Enabled" : "Disabled"]",
 			" ",
 			"Server Time: [time2text(world.timeofday, "YYYY-MM-DD hh:mm:ss")]",
-			"Station Time: [station_time_timestamp()]",
+			"Station Time: [SSday_night.get_twentyfourhour_timestamp()]",
```
Comment (azrael-x86) [inline code/controllers/subsystem/statpanel.dm:49]:
```suggestion
			"Station Time: [SSday_night.get_twentyfourhour_timestamp()]", // SKYRAT EDIT - DAY/NIGHT CYCLE
```

File: modular_skyrat/modules/day_night/code/day_night_timezones.dm [hunk 2]
```diff
@@ -0,0 +1,55 @@
+/**
+ * Timezones are simple presets to determine what color the sky will be at a certain time.
+ */
+/datum/timezone
+	/// The name of this timezone
+	var/name = "DEFALT"
```
Comment (azrael-x86) [inline modular_skyrat/modules/day_night/code/day_night_timezones.dm]:
```suggestion
	var/name = "DEFAULT"
```

File: code/controllers/subsystem/statpanel.dm (hunk 1 above)
Comment (Gandalf2k15) [reply code/controllers/subsystem/statpanel.dm:49]: already in a skyrat edit

File: modular_skyrat/modules/day_night/code/_day_night_defines.dm [hunk 3]
```diff
@@ -0,0 +1,17 @@
+/// The amount of minutes in an hour
+#define HOUR_INCREMENT 60
+/// The time at which  we reset to 0
+#define MIDNIGHT_RESET 24
+/// How many minutes we add to the time every subsystem fire
+#define SUBSYTEM_FIRE_INCREMENT 10
```
Comment (azrael-x86) [inline modular_skyrat/modules/day_night/code/_day_night_defines.dm:6]:
```suggestion
#define SUBSYSTEM_FIRE_INCREMENT 10
```
change all references too

**Pull request closed without merging**